File: ОП/18_04/4_пара/3z/3.cpp
/** Задача приветствует того, чьё имя вы введёте, только теперь чтение из файла и выводится в отдельный файл  */

// Проверил: Батраков Данила П1-18

#include <string>
#include <fstream> 
using namespace std;

int main(){
	string name;
	ifstream input("input.txt");
	input >> name;

	ofstream output("output.txt");
	output << "Hi, " << name << endl;

    return 0;
}

File: УП/К5.2/3.py
'''
Задание 3.
Разработать программу для нахождения наибольшего общего делителя
'''

#Алгоритм Евклида для нахождения НОД

a = int(input())
b = int(input())
 
while a!=0 and b!=0:
    if a > b:
        a = a % b
    else:
        b = b % a
 
print (a+b)
